spring-cloud-gcp-docs/src/main/asciidoc/spanner.adoc

```diff
@@ -885,6 +885,20 @@ point in time of the transaction.
 Because read-only transactions are non-locking and can be performed on points in time in the past, these
 are recommended for functions that do not perform write operations.
 
+===== @Transactional and declerative transaction management
+
+To use `@Transactional` in [declerative spring transaction management](https://docs.spring.io/spring/docs/current/spring-framework-reference/data-access.html#transaction-declarative)
+`SpannerTransactionManager` has to be registered in Spring.
+Example:
+```
+@Bean
+public SpannerTransactionManager transactionManager() {
+    return new SpannerTransactionManager(databaseClient)
```

Comment:
I think we should add it to our auto-configuration in `GcpSpannerAutoConfiguration`.

Reply:
Hm.. There is already Spring's annotation `@EnableTransactionManagement` that controls `@Transactional` support. 

Are you sure every developer would like to have transaction manager setup always on `@ GcpSpannerAutoConfiguration` ? I think more flexible to keep this as explicit config or dedicated annotation.

Reply:
Ah fair enough. Forgot about the fact that we already integrate with `@EnableTransactionManagement`.

Reply:
Or do we?

Reply:
`@EnableTransactionManagement` just enables `@Transactional` annotations. 
We don't add anything on top of `@EnableTransactionManagement`.

Reply:
I think not every consumer `@GcpSpannerAutoConfiguration` wants to have declarative transactions. In fact that aligned with Spring JDBC + Transaction management. 

In Spring JDBC you don't need to have transaction management - you can simply use JdbcTemplate. 

Transaction management enablement is covered by `@EnableTransactionManagement`and there is a spring doc section that covers requirements for transaction manager being registered to your datasource.

from developer experience i think we should keep it consistent. Developer tries to use Spanner - ok use SpannerTemplate, developer wants transactions - go read about spring transaction management and find there is `@transactional` annotation, developer annotates beans with it and realizes there is a requirement for transaction manager, developer goes to this repo and finds there is `SpannerTransactionManager`, developer registers it and now all works.

Reply:
@olivergierke Can you guide us on this please?

Reply:
Well, actually there is a `TransactionAutoConfiguration` which auto-configures for us the mentioned `@EnableTransactionManagement`. And this one depends on the `DataSourceTransactionManagerAutoConfiguration`, for example. This one, in turn, just register a `DataSourceTransactionManager`. The same happens in the `Neo4jDataAutoConfiguration`. 

So, for consistency with Spring Boot I would insist that we need to introduce a `SpannerTransactionManagerAutoConfiguration` - fully similar to that one `DataSourceTransactionManagerAutoConfiguration`. However we can do that in the separate issue.

Thanks.

Reply:
Yep, in a separate #1032.